### PR TITLE
Feat/otc workarounds1

### PR DESCRIPTION
**Pull request by garloff**
Note: These need some testing to ensure they don't break on other clouds.
The neutron.tf changes are in a patch file, thus harmless.
The changes to the cloud-init userdata to install docker-ce-cli and docker-compose
and the deinstallation of image-provided ansible *should* be harmless, but without
testing one never knows.

### CODE REVIEW

File: terraform/neutron-otc.patch [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
++  #availability_zone_hints = [var.network_availability_zone]
+ }
+ 
+ resource "openstack_networking_subnet_v2" "subnet_internal" {
+@@ -199,7 +199,7 @@ resource "openstack_networking_subnet_v2" "subnet_internal" {
+   cidr        = "192.168.32.0/20"
+   ip_version  = 4
+   gateway_ip  = null
+-  enable_dhcp = false
++  #enable_dhcp = false
```
Comment (berendt) [inline terraform/neutron-otc.patch:50]: Introduced a new variable enable_dhcp with https://github.com/osism/testbed/commit/b48fa7945498fb761089fc9ede119b3f6a189263. Please check if this works here.

File: terraform/manager.tf [hunk 2]
```diff
@@ -235,7 +235,11 @@ write_files:
       fi
     path: /root/manager.sh
     permissions: 0700
+packages:
+  - docker-ce-cli
+  - docker-compose
```
Comment (berendt) [inline terraform/manager.tf:240]: This is not necessary. The missing Docker resulted from a bug in the Bootstrap playbook.

https://github.com/osism/testbed/pull/546

File: terraform/manager.tf [hunk 3]
```diff
@@ -235,7 +235,11 @@ write_files:
...
 runcmd:
+  - "dpkg -r ansible || true"
```
Comment (berendt) [inline terraform/manager.tf:242]: Dito.

File: terraform/neutron-otc.patch [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
++  #availability_zone_hints = [var.network_availability_zone]
+ }
+ 
+ resource "openstack_networking_subnet_v2" "subnet_management" {
+@@ -180,7 +180,7 @@ resource "openstack_networking_subnet_v2" "subnet_management" {
+   network_id      = openstack_networking_network_v2.net_management.id
+   cidr            = "192.168.16.0/20"
+   ip_version      = 4
+-  dns_nameservers = ["8.8.8.8", "9.9.9.9"]
++  dns_nameservers = ["100.125.4.25", "9.9.9.9"]
```
Comment (berendt) [inline terraform/neutron-otc.patch:32]: I think it makes more sense to make this configurable via a new terraform variable.

**Review by berendt — CHANGES_REQUESTED** [review]

File: terraform/neutron-otc.patch (hunk 4 above)
Comment (berendt) [reply terraform/neutron-otc.patch:32]: Done. Is now configurable.

File: terraform/manager.tf (hunk 3 above)
Comment (garloff) [reply terraform/manager.tf:242]: Sure?
The OTC Ubuntu images come with ansible-2.9 preinstalled and that's a problem for the 2.10+ installation.
So I believe, this snippet will be needed in the future.

File: terraform/neutron-otc.patch (hunk 4 above)
Comment (garloff) [reply terraform/neutron-otc.patch:32]: > Done. Is now configurable.

Thanks!

File: terraform/manager.tf (hunk 2 above)
Comment (garloff) [reply terraform/manager.tf:240]: Hmm, I still saw a missing docker with recent testing ...

File: terraform/neutron-otc.patch (hunk 1 above)
Comment (garloff) [reply terraform/neutron-otc.patch:50]: It does, thanks!

**Comment by garloff** [comment]
A number of the required changes is superceeded by the new variables for dhcp and dns.
The rest will come in a separate new PR #594.

File: terraform/manager.tf (hunk 3 above)
Comment (berendt) [reply terraform/manager.tf:242]: Ok. Makes sense then. Please add as an independent PR.

File: terraform/manager.tf (hunk 2 above)
Comment (berendt) [reply terraform/manager.tf:240]: Then it is a bug elsewhere since Docker is installed by an Ansible role. If Docker is missing this role has failed.

Comment (garloff) [reply terraform/manager.tf:240]: ansible was failing on the OTC images due to ansible-2.9 being preinstalled there.
The dpkg -r ansible is all that's needed.

File: terraform/manager.tf (hunk 3 above)
Comment (garloff) [reply terraform/manager.tf:242]: This is currently in a separate patch file in PR #594 (ansible-otc.patch).
It's harmless on images where ansible is not preinstalled, but it's still an OTC specific workaround, so I wonder whether it's acceptable to have included always.

Comment (berendt) [reply terraform/manager.tf:242]: It does not interfere if Ansible is not installed and removes it if it is. Fits.

**Pull request closed without merging**